teuthology/provision/fog.py

```diff
@@ -292,7 +292,14 @@ def _wait_for_ready(self):
         sentinel_file = config.fog.get('sentinel_file', None)
         if sentinel_file:
             cmd = "while [ ! -e '%s' ]; do sleep 5; done" % sentinel_file
-            self.remote.run(args=cmd, timeout=600)
+            action = f"wait for sentinel on {self.shortname}"
+            with safe_while(action=action, timeout=1800, increment=3) as proceed:
+                while proceed():
+                    try:
+                        self.remote.run(args=cmd, timeout=600)
```

Comment:
How did you decide on these timeouts?

Overall LGTM, my comments are mainly trying to understand. Let's test it with the nop suite.

Reply:
The 1800 value came out of nowhere, really; accepting suggestions :)

Reply:
That's fine to start, was just curious how you decided on the value